include/envoy/network/post_io_action.h

```diff
@@ -10,7 +10,9 @@ enum class PostIoAction {
   // Close the connection.
   Close,
   // Keep the connection open.
-  KeepOpen
+  KeepOpen,
```

Comment:
Should this be a new value type with the following public api:

class PostIoAction {
public:
  // constructors
  static PostIoAction CloseGraceful();
  static PostIoAction CloseError();
  static PostIoAction keepOpen();

  bool shouldKeepOpen() const;
  bool errorDetected() const;
}

I think it would help ensure that Close requests are handled correctly.